build-logic/gradle/src/main/kotlin/com/avito/android/GenerateCommonProperties.kt

```diff
@@ -0,0 +1,55 @@
+package com.avito.android
+
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputDirectories
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+abstract class GenerateCommonProperties : DefaultTask() {
+
+    @get:InputFile
+    abstract val commonPropertiesFile: RegularFileProperty
+
+    @get:OutputDirectories
+    abstract val projectDirs: SetProperty<Directory>
+
+    @TaskAction
+    fun doWork() {
+        val commonProperties = loadProperties(commonPropertiesFile.get().asFile)
+
+        projectDirs.get().forEach { projectDir ->
```

Comment:
ProjectDirs - is Gradle fun?

Reply:
nope, it's task property, task output specifically

Reply:
Sorry, my eyes was broken ))